backend/organization/views/project_views.py

```diff
@@ -23,14 +37,102 @@ def get_serializer_class(self):
         return ProjectMinimalSerializer
 
 
-class ProjectAPIView(ListAPIView):
-    lookup_field = 'pk'
-    serializer_class = ProjectSerializer
-    pagination_class = None
+"""
+Note: I am splitting organization project creation and personal project creation into
+two separate APIs. It makes more sense do two APIs so that we can manage user accessibility
+by project or organization member roles.
+"""
+
+
+class OrganizationProjectsView(APIView):
+    permission_classes = [OrganizationProjectCreationPermission]
+
+    def post(self, request, organization_id):
+        organization = check_organization(organization_id)
+        if not organization:
+            return Response({
+                'message': 'Organization not found'
+            }, status=status.HTTP_404_NOT_FOUND)
+        project_creation_pre_checks(request.data)
+        project = create_new_project(request.data)
+
+        if 'order' in request.data:
+            order = request.data['order']
+            ProjectParents.objects.create(
+                project=project, parent_organization=organization,
+                parent_user=request.user, order=order
+            )
+
+        serializer = ProjectSerializer(project)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CreateProjectView(APIView):
     permission_classes = [IsAuthenticated]
 
-    def get_queryset(self):
-        return Project.objects.filter(id=int(self.kwargs['pk']))
+    def post(self, request):
+        project_creation_pre_checks(request.data)
+        project = create_new_project(request.data)
+        if 'order' in request.data:
+            order = request.data['order']
+            ProjectParents.objects.create(
+                project=project, parent_user=request.user, order=order
+            )
+
+        # Here a user is creating an individual project so we will be adding them as a project member
+        role = Role.objects.get(name="Project Administrator")
+        ProjectMember.objects.create(
+            project=project, user=request.user, role=role
+        )
+        serializer = ProjectSerializer(project)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class ProjectAPIView(APIView):
+    permission_classes = [OrganizationProjectCreationPermission]
```

Comment:
Does setting the permission to this mean that users who don't fulfil this permission also can't do the get request to view a project?

Reply:
Anyone who is authenticated can read a project. Only people with the permission will be able to edit project.